src/Event/Http/HttpRequestEvent.php

```diff
@@ -171,13 +171,19 @@ private function rebuildQueryString(): string
     {
         if ($this->payloadVersion === 2.0) {
             $queryString = $this->event['rawQueryString'] ?? '';
-            // We re-parse the query string to make sur it is URL-encoded
+            // We re-parse the query string to make sure it is URL-encoded
             // Why? To match the format we get when using PHP outside of Lambda (we get the query string URL-encoded)
             parse_str($queryString, $queryParameters);
             return http_build_query($queryParameters);
         }
 
+        // It is likely that we do not need to differentiate between API Gateway (Version 1) and ALB. However,
+        // it would lead to a breaking change since the current implementation for API Gateway does not
+        // support MultiValue query string. This way, the code is fully backward-compatible while
+        // offering complete support for multi value query parameters on ALB. Later on there can
+        // be a feature flag that allows API Gateway users to opt-in to complete support as well.
```

Comment:
Ohh that's interesting, so you mean that maybe we can use the same logic with API Gateway to support multivalue query string parameters ? That's not our goal in this PR, but that's useful for later, thanks!

Reply:
I will make some time in the coming days to test my project behind API Gateway just to confirm this and if I find anything that invalidate my comment, I'll send a PR to change it, but I think this is correct because A) both ALB and API Gateway uses the same attribute name and B) they both use the same strategy to always send single value or always-array depending on 'multi-value' flag.